dll/win32/shlwapi/ordinal.c

```diff
@@ -5350,6 +5350,35 @@ HRESULT VariantChangeTypeForRead(_Inout_ VARIANTARG *pvarg, _In_ VARTYPE vt)
     return hr;
 }
 
+BOOL
+VariantToBuffer(
```

Comment:
More like "VariantArrayToBuffer" (since it's only dealing with that?)
If this is really so, it seems you could do very simply:
```c
    /* Only supports byte array */
    if (!varIn || V_VT(varIn) != (VT_UI1 | VT_ARRAY))
        return FALSE;

    /* Boundary check and access */
    // Continue here as usual...
    return TRUE;
```

Reply:
Improved.